Getregion opts (#14090)
vim/vim: runtime/doc/builtin.txt

getreg([{regname} [, 1 [, {list}]]])
				String or List   contents of a register
getreginfo([{regname}])		Dict	information about a register

> getregion({pos1}, {pos2}[, {opts}])

Reviewer: ```suggestion
getregion({pos1}, {pos2} [, {opts}])
```

		Can also be used as a |method|: >
			GetRegname()->getreginfo()

getregion({pos1}, {pos2}, {type} [, {opts}])			*getregion()*
		Returns the list of strings from {pos1} to {pos2} as if it's
		selected in visual mode of {type}.

		For possible values of {pos1} and {pos2} are |List| or |string|.
		When {pos1} or {pos2} is |string|, see |line()|.

> 		When {pos1} or {pos2} is |List|, with two, three or four item:

Reviewer: ```suggestion
		When {pos1} or {pos2} is |List|, with two or three items:
```

		Can also be used as a |method|: >
			GetRegname()->getreginfo()

getregion({pos1}, {pos2} [, {opts}])			*getregion()*
		Returns the list of strings from {pos1} to {pos2} in current
		buffer.

		{pos1} and {pos2} must be a |List| with four numbers:
			[bufnum, lnum, col, off]
		"bufnum" is zero, unless a mark like '0 or 'A is used, then it

> 		is the buffer number of the mark.

Reviewer: There is no mention of "mark" anywhere else in the docs of this function, and it doesn't allow non-current buffer.

		"lnum" and "col" are the position in the buffer.  The first
		column is 1.

> 		The "off" number is ignored.

Reviewer: I don't think "off" is ignored. `coladd` is used in the code.

Author: Oh, I forgot it.

		Can also be used as a |method|: >
			GetRegname()->getreginfo()

getregion({pos1}, {pos2} [, {opts}])			*getregion()*
		Returns the list of strings from {pos1} to {pos2} in current
		buffer.

> 		{pos1} and {pos2} must be a |List| with four numbers:
> 			[bufnum, lnum, col, off]
> 		"bufnum" must be zero or current buffer number.
> 		"lnum" and "col" are the position in the buffer.  The first
> 		column is 1.
> 		The "off" number is zero, unless 'virtualedit' is used.  Then
> 		it is the offset in screen columns from the start of the
> 		character.  E.g., a position within a <Tab> or after the last
> 		character.

Reviewer: I think it's better to simply reference `getpos()`
```suggestion
		{pos1} and {pos2} must be a |List| with four numbers in the
		same format as the result of |getpos()|:
			[bufnum, lnum, col, off]
		While "bufnum" must be zero or current buffer number.
```

		Can also be used as a |method|: >
			GetRegname()->getreginfo()

> getregion({pos1}, {pos2}[, {opts}])			*getregion()*

Reviewer: ```suggestion
getregion({pos1}, {pos2} [, {opts}])			*getregion()*
```

		Returns the list of strings from {pos1} to {pos2} in current
		buffer.

		{pos1} and {pos2} must be a |List| with four numbers:
			[bufnum, lnum, col, off]
		"bufnum" is zero, unless a mark like '0 or 'A is used, then it
		is the buffer number of the mark.
		"lnum" and "col" are the position in the buffer.  The first
		column is 1.
		The "off" number is zero, unless 'virtualedit' is used.  Then
		it is the offset in screen columns from the start of the
		character.  E.g., a position within a <Tab> or after the last
		character.

> 		{type} is the selection type:

Reviewer 2: Should this line be removed?

Author: Yes.  I have removed.

		- If the selection starts or ends in the middle of a multibyte
		  character, it is not included but its selected part is
		  substituted with spaces.

> 		- If {type} is an invalid string, an empty list is returned.

Reviewer 2: `{type}` is in the dict now, and here it is referenced as a standalone parameter of a function.

		- If {pos1} or {pos2} is not current buffer, an empty list is
		  returned.

		Examples: >
			:xnoremap <CR>
			\ <Cmd>echow getregion(
			\ getpos('v'), getpos('.'), #{ type: mode() })<CR>
<
		Can also be used as a |method|: >

> 			'.'->getregion(getpos("'a"))

Reviewer 2: In this example '.' is a string representing current position? But pos1 couldn't be a string, right?

Author: Fixed.
